File: backend/app/models/event.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field

class ObservationPoint(BaseModel):
    time: str
    frp: float
    confidence: Optional[float] = None
    brightness: Optional[float] = None
    lat: Optional[float] = None
    lon: Optional[float] = None

class EvidenceItem(BaseModel):
    type: str
    label: str
    weight: str = "high"  # low, medium, high

class PriorityBreakdown(BaseModel):
    thermal_severity: float = Field(..., description="Weight 30%")
    anomaly_score: float = Field(..., description="Weight 25%")
    persistence_score: float = Field(..., description="Weight 20%")
    exposure_score: float = Field(..., description="Weight 15%")
    industrial_relevance: float = Field(..., description="Weight 10%")

class ProbabilityItem(BaseModel):
    category: str
    probability: float

class EventSummary(BaseModel):
    id: str
    lat: float
    lon: float
    first_seen: str
    last_seen: str
    observation_count: int = 1
    peak_frp: float
    mean_frp: float
    persistence_hours: float
    probable_source: str
    confidence: float
    baseline_frp: float
    anomaly_ratio: float
    anomaly_score: float
    exposure_score: float = 50.0
    industrial_relevance: float = 50.0
    priority_score: float
    priority_level: str  # LOW, MODERATE, HIGH, CRITICAL
    nearest_facility: str
    facility_type: str = "Industrial Facility"
    facility_distance_km: float
    land_cover: str
    status: str = "REVIEW"  # REVIEW, MONITOR, ESCALATED, CONFIRMED, FALSE_POSITIVE

class EventDetail(EventSummary):
    evidence: List[str] = []
    evidence_items: List[EvidenceItem] = []
    timeline: List[ObservationPoint] = []
    probabilities: List[ProbabilityItem] = []
    priority_breakdown: Optional[PriorityBreakdown] = None
    context: Dict[str, Any] = {}
    caveats: List[str] = [
        "Thermal observations alone do not confirm source type.",
        "Classification represents evidence-weighted probable source, not ground-truth confirmation.",
        "Thresholds and scores are prototype engineering heuristics."
    ]

class AnalyticsSummary(BaseModel):
    observations: int
    events: int
    industrial_candidates: int
    high_critical: int
    by_source: Dict[str, int]
    by_priority: Dict[str, int]
    avg_persistence_hours: float
    peak_frp_max: float

class SourceItemStatus(BaseModel):
    name: str
    purpose: str
    mode: str  # "demo" or "live"
    status: str  # "healthy", "connected", "fallback"
    last_update: Optional[str] = None
    details: Optional[str] = None

class SourcesStatus(BaseModel):
    firms: SourceItemStatus
    osm: SourceItemStatus
    worldcover: SourceItemStatus
    historical: SourceItemStatus
    copernicus: SourceItemStatus

class FeedbackRequest(BaseModel):
    action: str  # review, confirm, false_positive, escalate
    note: Optional[str] = None

class IngestRequest(BaseModel):
    bbox: Optional[List[float]] = [68.0, 8.0, 97.5, 37.0]
    days: Optional[int] = 1

File: frontend/src/types/event.ts
export interface ObservationPoint {
  time: string;
  frp: number;
  confidence?: number;
  brightness?: number;
  lat?: number;
  lon?: number;
}

export interface EvidenceItem {
  type: string;
  label: string;
  weight: 'low' | 'medium' | 'high';
}

export interface PriorityBreakdown {
  thermal_severity: number;
  anomaly_score: number;
  persistence_score: number;
  exposure_score: number;
  industrial_relevance: number;
}

export interface ProbabilityItem {
  category: string;
  probability: number;
}

export interface EventSummary {
  id: string;
  lat: number;
  lon: number;
  first_seen: string;
  last_seen: string;
  observation_count: number;
  peak_frp: number;
  mean_frp: number;
  persistence_hours: number;
  probable_source: string;
  confidence: number;
  baseline_frp: number;
  anomaly_ratio: number;
  anomaly_score: number;
  exposure_score: number;
  industrial_relevance: number;
  priority_score: number;
  priority_level: 'LOW' | 'MODERATE' | 'HIGH' | 'CRITICAL';
  nearest_facility: string;
  facility_type: string;
  facility_distance_km: number;
  land_cover: string;
  status: 'REVIEW' | 'MONITOR' | 'ESCALATED' | 'CONFIRMED' | 'FALSE_POSITIVE';
}

export interface EventDetail extends EventSummary {
  evidence: string[];
  evidence_items?: EvidenceItem[];
  timeline: ObservationPoint[];
  probabilities: ProbabilityItem[];
  priority_breakdown?: PriorityBreakdown;
  context: {
    nearest_facility?: string;
    facility_type?: string;
    facility_distance_km?: number;
    land_cover?: string;
    transport_access?: string;
    demographic_exposure?: string;
    [key: string]: any;
  };
  caveats: string[];
}

export interface AnalyticsSummary {
  observations: number;
  events: number;
  industrial_candidates: number;
  high_critical: number;
  by_source: Record<string, number>;
  by_priority: Record<string, number>;
  avg_persistence_hours: number;
  peak_frp_max: number;
}

export interface SourceItemStatus {
  name: string;
  purpose: string;
  mode: string;
  status: string;
  last_update?: string;
  details?: string;
}

export interface SourcesStatus {
  firms: SourceItemStatus;
  osm: SourceItemStatus;
  worldcover: SourceItemStatus;
  historical: SourceItemStatus;
  copernicus: SourceItemStatus;
}
